txnkv/txnsnapshot/snapshot.go

```diff
@@ -623,12 +600,29 @@ func (s *KVSnapshot) Get(ctx context.Context, k []byte) ([]byte, error) {
 		}
 	}(time.Now())
 
+	s.mu.RLock()
+	// Check the cached values first.
+	if s.mu.cached != nil {
+		if value, ok := s.mu.cached[string(k)]; ok {
+			atomic.AddInt64(&s.mu.hitCnt, 1)
+			s.mu.RUnlock()
+			if len(value) == 0 {
+				return nil, tikverr.ErrNotExist
+			}
+			return value, nil
+		}
```

Comment:
The `s.mu` is not released in the else branch.

Reply:
It's released in [L632](https://github.com/tikv/client-go/blob/90f9b060de894153ace9e6c16be1d79efe6d7c57/txnkv/txnsnapshot/snapshot.go#L632)